nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/PutAzureDataLakeStorage.java

```diff
@@ -120,11 +122,29 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
 
                 final long length = flowFile.getSize();
                 if (length > 0) {
-                    try (final InputStream rawIn = session.read(flowFile); final BufferedInputStream in = new BufferedInputStream(rawIn)) {
-                        fileClient.append(in, 0, length);
+                    long chunkStart = 0;
+                    long chunkSize;
+
+                    try (final InputStream rawIn = session.read(flowFile);
+                         final BufferedInputStream in = new BufferedInputStream(rawIn) {
+                             @Override
+                             public int available() {
+                                 // com.azure.storage.common.Utility.convertStreamToByteBuffer() throws an exception
+                                 // if there are more available bytes in the stream after reading the chunk
+                                 return 0;
```

Comment:
@MuazmaZ Do you happen to know why `Utility.convertStreamToByteBuffer()` throws an exception when `available() > 0`?
https://github.com/Azure/azure-sdk-for-java/blob/0345889402425191b7003e73b7b3d6ea3c0a5175/sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/Utility.java#L268

Due to this, it is not possible to process a longer input stream in portions / chunks.
As a workaround, I added a fake `available()` method to lie there is no more data in the input stream which is not really nice but works.
Another option would be to read the chunks in a loop into a byte array on our side and pass a stream on the byte array to the Azure client lib. But I would rather avoid this extra copy and extra memory for the buffer.

Reply:
@turcsanyip , the patch looks good to me, but maybe you could use the `BoundedInputStream` from the Apache Commons library [1] instead of the workaround. What do you think?
[1] https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/input/BoundedInputStream.html

Reply:
@adenes Thanks for the idea. `BoundedInputStream` works properly here.

Reply:
@turcsanyip I am looking into this and I will respond by tomorrow based on the internal team's response.

Reply:
@MuazmaZ Thanks. The `BoundedInputStream` approach is much better, than my original workaround, so it is not so critical anymore. However, I'm still wondering why it is not possible to pass in a longer stream to `Utility.convertStreamToByteBuffer()`.